### ISSUE TITLE
Crash when voyager-operator starts. Observed a panic: "invalid memory address or nil pointer dereference"

### ISSUE THREAD

**Issue by kksudo** [voyagermesh/voyager#1515]
### Env
voyager: `v12.0.0`
[helm chart prometheus-operator](https://github.com/helm/charts/tree/master/stable/prometheus-operator): `8.13.14`
[prometheus-operator](https://github.com/coreos/prometheus-operator): `0.38.1` or `0.34.0`
k8s: `Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.11-gke.12"`
helm: `2.14.1`

### Description
The voyager version `v12.0.0` and higher doesn't working (CrashloopBackOff) with crd `prometheuses.monitoring.coreos.com`. If remove the crd `prometheuses.monitoring.coreos.com`,  voyager start working normally. But the version `v11.0.1` working good with both (`0.38.1` or `0.34.0`) version of crd prometheus.

### Logs
<details>
  <summary>Voyager operator logs</summary>
 
  ```
E0616 23:54:01.218006       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 1016 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x2606140, 0x44cfb00)
	/src/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/src/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x2606140, 0x44cfb00)
	/usr/local/go/src/runtime/panic.go:969 +0x166
kmodules.xyz/monitoring-agent-api/agents/coreosprometheusoperator.(*PrometheusCoreosOperator).CreateOrUpdate(0xc003176900, 0x2ff45e0, 0xc00117a000, 0xc0031675c0, 0x0, 0xc001456800, 0x40c7a6, 0xc0030530c8)
	/src/vendor/kmodules.xyz/monitoring-agent-api/agents/coreosprometheusoperator/lib.go:73 +0x329
voyagermesh.dev/voyager/pkg/ingress.(*controller).ensureMonitoringAgent(0xc000caef20, 0xc0031675c0, 0x445ebc1, 0x0, 0x0, 0x0)
	/src/pkg/ingress/monitor.go:44 +0x1fe
voyagermesh.dev/voyager/pkg/ingress.(*loadBalancerController).Reconcile(0xc00000ed30, 0x7, 0xc000b02f60)
	/src/pkg/ingress/loadbalancer.go:217 +0x503
voyagermesh.dev/voyager/pkg/operator.(*Operator).reconcileEngress(0xc000b70380, 0xc00097f1e0, 0x1f, 0xc0005ba3a0, 0xc00076b500)
	/src/pkg/operator/ingress_crds.go:135 +0x8e2
kmodules.xyz/client-go/tools/queue.(*Worker).processNextEntry(0xc000567500, 0xc00076b700)
	/src/vendor/kmodules.xyz/client-go/tools/queue/worker.go:84 +0xec
kmodules.xyz/client-go/tools/queue.(*Worker).processQueue(0xc000567500)
	/src/vendor/kmodules.xyz/client-go/tools/queue/worker.go:67 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000ec9cf0)
	/src/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5f
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000ec9cf0, 0x3b9aca00, 0x0, 0x1, 0xc000114360)
	/src/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000ec9cf0, 0x3b9aca00, 0xc000114360)
	/src/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by kmodules.xyz/client-go/tools/queue.(*Worker).Run
	/src/vendor/kmodules.xyz/client-go/tools/queue/worker.go:53 +0x89
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2195fa9]

goroutine 1016 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/src/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x105
panic(0x2606140, 0x44cfb00)
	/usr/local/go/src/runtime/panic.go:969 +0x166
kmodules.xyz/monitoring-agent-api/agents/coreosprometheusoperator.(*PrometheusCoreosOperator).CreateOrUpdate(0xc003176900, 0x2ff45e0, 0xc00117a000, 0xc0031675c0, 0x0, 0xc001456800, 0x40c7a6, 0xc0030530c8)
	/src/vendor/kmodules.xyz/monitoring-agent-api/agents/coreosprometheusoperator/lib.go:73 +0x329
voyagermesh.dev/voyager/pkg/ingress.(*controller).ensureMonitoringAgent(0xc000caef20, 0xc0031675c0, 0x445ebc1, 0x0, 0x0, 0x0)
	/src/pkg/ingress/monitor.go:44 +0x1fe
voyagermesh.dev/voyager/pkg/ingress.(*loadBalancerController).Reconcile(0xc00000ed30, 0x7, 0xc000b02f60)
	/src/pkg/ingress/loadbalancer.go:217 +0x503
voyagermesh.dev/voyager/pkg/operator.(*Operator).reconcileEngress(0xc000b70380, 0xc00097f1e0, 0x1f, 0xc0005ba3a0, 0xc00076b500)
	/src/pkg/operator/ingress_crds.go:135 +0x8e2
kmodules.xyz/client-go/tools/queue.(*Worker).processNextEntry(0xc000567500, 0xc00076b700)
	/src/vendor/kmodules.xyz/client-go/tools/queue/worker.go:84 +0xec
kmodules.xyz/client-go/tools/queue.(*Worker).processQueue(0xc000567500)
	/src/vendor/kmodules.xyz/client-go/tools/queue/worker.go:67 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000ec9cf0)
	/src/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5f
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000ec9cf0, 0x3b9aca00, 0x0, 0x1, 0xc000114360)
	/src/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000ec9cf0, 0x3b9aca00, 0xc000114360)
	/src/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by kmodules.xyz/client-go/tools/queue.(*Worker).Run
	/src/vendor/kmodules.xyz/client-go/tools/queue/worker.go:53 +0x89
  ```
  
</details>

**Comment by kksudo** [comment]
Seems like a problem with **servicemonitor**. If the **servicemonitor** turned off, then the voyager operator 12 work fine.

**Comment by rechandler12** [comment]
I have the same issue

**Comment by sniip-code** [comment]
same issue here went back to v11.01

**Comment by endrec** [comment]
same issue, 12.0.0-rc.1 works fine, but rc.2 and above crashes.

**Comment by artazar** [comment]
Same issue with the released 12.0.0.

Tried starting voyager operator with v13.0.0-beta.1 image tag, no crashes occur, but there are errors:

`E0114 01:49:37.523861       1 reflector.go:178] k8s.io/client-go/informers/factory.go:135: Failed to list *v1beta1.Ingress: ingresses.networking.k8s.io is forbidden: User "system:serviceaccount:voyager:voyager" cannot list resource "ingresses" in API group "networking.k8s.io" at the cluster scope`

(Looks like the next version is going to move from CRD to the stock ingresses.networking.k8s.io API object support.)

It would be very nice to resolve this issue in 12.0 version, because this effectively leaves the ingress without proper monitoring.

There is even no w/a is to manage the servicemonitor object separately, as exposing the prometheus formatted stats is dependent on the ingress annotations that lead to the crash.

**Comment by tamalsaha** [comment]
Reopen if you still face this issue in Voyager v2021.10.18
